common/build/eslint-config-fluid/package.json

```diff
@@ -36,6 +36,6 @@
     "eslint": "~8.6.0",
     "node-jq": "^2.3.3",
     "sort-json": "^2.0.1",
-    "typescript": "^4.5.3"
```

Comment:
what's the plan around the version pinning here?  is the tilde change permanent and generally to force more rigidity?

Reply:
everywhere else in the repo (or at least most other places) use "~" with typescript, so this just makes them all consistent.
I think using "~" is the right choice as minor version updates to typescript can require code changes (as shown in this change), especially in you use strict (which we often do)

This particular package though is special in that it doesn't use typescript, and really only needs "typescript@>=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev || >= 3.4.0-dev || >= 3.5.0-dev || >= 3.6.0-dev || >= 3.6.0-beta || >= 3.7.0-dev || >= 3.7.0-beta" due to peer dependency.

That said, testing our linter config with the same version of typescript it will be used with seems like a good idea, and version consistency is nice (makes future upgrades easier among other things) so I think making this match the rest of the packages seems like a good idea, but I don't feel strongly either way.

Reply:
Also its a dev dependency, so it doesn't impact consumers of this package, just us: were it a prod dep, I'd suggest relaxing it.

Reply:
For posterity, we also use tilde dependencies with eslint, for similar reasons. This is going on my list of policies we could/should enforce.